### PR TITLE
Remove redundant else block

### PR DESCRIPTION
**Pull request by meinac**
### Summary

Remove redundant `else; nil; end` block. **if block** already returns nil in case it does not branch.

### CODE REVIEW

**Comment by meinac** [comment]
cool, I did not search the pattern in the project. I was just wandering around and saw it. I'll search the pattern in entire project and address them at once.

**Comment by meinac** [comment]
Thanks again for pointing this out. I run the regex `else(?:\n|;)\s*nil(?:\n|;)\s*end` to find all the occurrences of redundant else blocks and removed them.

> This seems like a good candidate for a Rubocop rule!

Completely agree, lets see what the core team members say.

**Comment by composerinteralia** [comment]
Try [this RuboCop rule](https://github.com/bbatsov/rubocop/blob/master/manual/cops_style.md#styleemptyelse).

You can add this to the `.rubocop.yml`:

```yml
Style/EmptyElse:
  EnforcedStyle: both
```

and then let `rubocop -a` fix any remaining violations.

**Comment by matthewd** [comment]
Hmm... I'm not sure about this.

I know they're the same, and I trust a future reader to know they're the same. But it feels like the author has stopped in the middle of.

It's totally unnecessary for the intended meaning, but omitting that last part still feels like bad writing. (In a value context, where the fact it evaluates to nil is directly relevant / significant to whatever's using the value, and not just a throw-away "well we can't do anything better".)

File: activestorage/lib/active_storage/attached.rb [hunk 1]
```diff
@@ -28,8 +28,6 @@ def create_blob_from(attachable)
           ActiveStorage::Blob.create_after_upload!(attachable)
         when String
           ActiveStorage::Blob.find_signed(attachable)
-        else
-          nil
```
Comment (claudiob) [inline activestorage/lib/active_storage/attached.rb]: The presence of this `else` was previously discussed in this comment:
https://github.com/rails/rails/commit/b7cc003aa0aada594cb18ab80c13c13c75bcd389#r27841179

**Review by sikachu — COMMENTED** [review]
Per @claudiob's comment, I think @dhh has made it clear that these are being intentionally left in. I think I'd let him make a call if we should merge/close this PR.

**Comment by dhh** [comment]
These are indeed all intentional, and I think better writing, as Matthew says. Having to carry in your head the implicit return value of nil when a branch isn’t found is needlessly taxing. A conditional tree should present all the possible options exhaustively.

Thanks for starting the discussion though!

**Pull request closed without merging**